backend/app/controllers/users_controller.rb

```diff
@@ -4,12 +4,39 @@ class UsersController < ApplicationController
   def index
     @users = policy_scope(User).order(:email)
 
-    if users_with_roles_only?
+    users_with_roles_only = params.fetch(:users_with_roles_only, nil)
+
+    if users_with_roles_only == "true"
       @users = @users.with_at_least_one_role
     end
 
+    filter = params.fetch(:filter, nil)
+    puts filter.inspect
+
+    if filter.present?
+      begin
+        filter_email = JSON.parse(filter)
+        filter_email.each do |email|
+          filter_value = email.values.join(" , ")
+          sanitized_filter_value = Regexp.escape(filter_value)
+          san_fil_val_without_accent = ActiveSupport::Inflector.transliterate(sanitized_filter_value, " ")
+          next if san_fil_val_without_accent == ""
+
+          @users = @users.where("email like ?", "#{san_fil_val_without_accent}%")
```

Comment:
you should keep the comparison with *~ instead of like, as the string is escaped to be used as a regexp not as an operand of "like".